src/peergos/server/tests/UserTests.java

```diff
@@ -583,16 +583,21 @@ public void cleanupFailedUpload() throws Exception {
                         context.crypto.random)).join();
         int prior = context.getTotalSpaceUsed(context.signer.publicKeyHash, context.signer.publicKeyHash).get().intValue();
 
-        context.getTransactionService().open(transaction).join();
+        TransactionService transactions = context.getTransactionService();
+        network.synchronizer.applyComplexUpdate(userRoot.owner(), transactions.getSigner().join(),
+                (s, committer) -> transactions.open(s, committer, transaction)).join();
         try {
             userRoot.uploadOrOverwriteFile(filename, throwingReader, data.length, context.network,
                     context.crypto, l -> {}, transaction.getLocations()).get();
         } catch (Exception e) {}
         int during = context.getTotalSpaceUsed(context.signer.publicKeyHash, context.signer.publicKeyHash).get().intValue();
         Assert.assertTrue("One chunk uploaded", during > 5 * 1024*1024);
 
-        Set<Transaction> pending = context.getTransactionService().getOpenTransactions().join();
-        pending.forEach(t -> context.getTransactionService().clearAndClose(t).join());
+        network.synchronizer.applyComplexUpdate(userRoot.owner(), transactions.getSigner().join(),
+                (s, committer) -> {
+                    Set<Transaction> pending = transactions.getOpenTransactions(s).join();
+                    return Futures.reduceAll(pending, s, (v, t) -> transactions.clearAndClose(v, committer, t), (a, b) -> b);
```

Comment:
too many single letter vars. perhaps make t -> txn and v -> snapshotVersion

Reply:
fixed